### ISSUE TITLE
MixedLM Predict Clarification

### ISSUE THREAD

**Issue by jkelley7** [statsmodels/statsmodels#7065]
As I was going thru the documentation of MixedLM, I noticed the predict method only returns the fixed effects mean structure of the model. I was curious as to why this is the case. Shouldn't it return the fixed and random affects of the fitted model?

When going over the fittedvalues method it returns both the fixed effects and the predicted random effects. I guess i'm unclear as to why the predict method doesn't include predicted random effects.

I'm fairly new to Mixed models and am teaching myself via the Multilevel modeling in Plain Language book.

**Comment by kshedden** [comment]
Both are useful.  The docstrings seem to correctly describe what you get.

`fittedvalues` needs to be cached so has to remain as-is.

We could add a keyword parameter to `predict` giving the option to include the BLUPs of the random effects.

**Comment by jkelley7** [comment]
I understand the [docstring](https://www.statsmodels.org/dev/_modules/statsmodels/regression/mixed_linear_model.html#MixedLM.predict) and it is clear that ONLY the fixed effects mean structure is returned. My question is why only return the fixed effects?

Maybe i'm reading the wrong docstring but where does it say you can add in the random effects?

**Comment by humana** [comment]
Is there example code anywhere I could look at to include the BLUP of the random effects? I'm trying to create my own predict method to this, but I'm having a very difficult time without modifying the statsmodels source code.

**Comment by kshedden** [comment]
Below is an example, is this what you are trying to do?

`r.random_effects` is a dictionary that maps group labels to their random effect BLUPs.  So you just have to re-align that to the group labels.

```
import numpy as np
import statsmodels.api as sm
import pandas as pd

n = 100
df = pd.DataFrame({"x1": np.random.normal(size=n), "x2": np.random.normal(size=n)})
df["g"] = np.kron(np.arange(10), np.ones(10)).astype(np.int)
df["y"] = np.random.normal(size=n) + np.random.normal(size=10)[df["g"]]

m = sm.MixedLM.from_formula("y ~ x1 + x2", groups="g", data=df)
r = m.fit()

re = r.random_effects
rev = [re[g]["g"] for g in df["g"]]

y_hat = r.predict() + rev
```

**Comment by humana** [comment]
Ah yeah that's exactly what I was looking for, thanks.  I think it would be super useful if there was a parameter to the .predict() method to do this.

**Comment by huipingDS** [comment]
Yes, I agreed. Ran into the same problem when trying to use the model to make predictions on new data (except my model also has a different slop for each group). Anxiously waiting on being able to include random effects in .predict()!

**Comment by nikunjtiwari9** [comment]
Hi there , can you please help me to calculate the predicted value for this scenario.

vc = {"g1": "0 + C(group1)", "g2": "0 + C(group2)"}
oo = np.ones(df.shape[0])
model3 = sm.MixedLM.from_formula("y ~ 1", groups=oo, vc_formula=vc, data=df)
result3 = model3.fit()
